src/item_location.cpp

```diff
@@ -243,12 +284,36 @@ class item_location::item_on_vehicle : public item_location::impl
         vehicle_cursor cur;
 
     public:
-        item_on_vehicle( const vehicle_cursor &cur, item *which ) : cur( cur ) {
-            if( !( cur.has_item( *which ) || &cur.veh.parts[ cur.part ].base == which ) ) {
-                debugmsg( "Cannot locate item on vehicle: %s", cur.veh.name.c_str() );
-            } else {
-                what = which;
+        item_on_vehicle( const vehicle_cursor &cur, item_uid &&uid )
+            : impl( std::move( uid ) ), cur( cur ) {}
+
+        void serialize( JsonOut &js ) const override {
+            js.start_object();
+            js.member( "type", "vehicle" );
+            js.member( "uid", uid );
+            js.member( "position", position() );
+            js.member( "part", cur.part );
+            js.end_object();
+        }
+
+        item *target() const override {
+            if( what ) {
+                return what;
             }
+
+            /** as a special case item_location can refer to vehicle parts */
+            if( cur.veh.parts[ cur.part ].base.uid == uid ) {
```

Comment:
Probably should become it's own specialist type of `item_location` eventually, possibly along with furniture. Those two combined would allow us to get rid of a lot of the `pseudo_` item hacks we are currently using.